Spanner/tests/Perf/Operation.php

```diff
@@ -0,0 +1,267 @@
+        $this->weights = $weights;
+        $this->operations = $operations;
+        $this->latency = $latency;
+    }
+
+    public function load()
+    {
+        $this->keys = [];
+
+        $startTime = microtime(true);
+        $snapshot = $this->database->snapshot();
+
+        // Kind of assuming that id is always name of PK in whatever table you choose
+        $results = $snapshot->execute('SELECT id FROM ' . $this->parameters['table']);
+        foreach ($results as $row) {
+            $this->keys[] = $row['id'];
+        }
+
+        return microtime(true) - $startTime;
+    }
+
+    /**
+     * Run a single thread of the workload
+     *
+     * @return void
+     */
+    public function run()
+    {
+        if (!is_array($this->keys)) {
+            throw new \RuntimeException('Call Operation::load() first.');
+        }
+
+        $operationCount = (int) $this->parameters['operationcount'];
+        for ($i = 0; $i < $operationCount; $i++) {
+            $weight = (float) rand(0, $this->totalWeight * 10000) / 10000.0;
+
+            for ($j = 0; $j < count($this->weights); $j++) {
+                if ($weight <= $this->weights[$j]) {
+                    $operation = $this->operations[$j];
+
+                    $this->latency[$operation][] = $this->runOperation(
+                        $this->database,
+                        $this->parameters['table'],
+                        $operation
+                    );
+                    break;
+                }
+            }
+        }
+
+        return $this->latency;
+    }
+
+    /**
+     * Executes an operation.
+     *
+     * @param string $database
+     * @param string $table
+     * @param array $operation
+     * @return void
+     */
+    private function runOperation($database, $table, $operation)
+    {
+        $key = $this->keys[array_rand($this->keys)];
+        switch ($operation) {
+            case 'read':
+                $optime = $this->performRead($database, $table, $key);
+                break;
+            case 'update':
+                $optime = $this->update($database, $table, $key);
+                break;
+            case 'insert':
+                $optime = $this->insert($database, $table);
+                break;
+            case 'scan':
+                $optime = $this->scan($database, $table, $key);
+                break;
+        }
+
+        return $optime;
+    }
+
+    /**
+     * Executes a read.
+     *
+     * @param string $database
+     * @param string $table
+     * @param string $key
+     * @return float Time elapsed.
+     */
+    private function performRead($database, $table, $key)
+    {
+        $startTime = microtime(true);
+
+        // Kind of assuming that id is ubiquitous...
+        $results = $database->execute('SELECT * FROM ' . $table . ' where id = @id', [
+            'parameters' => [
+                'id' => $key
+            ]
+        ]);
+
+        foreach ($results as $row) {
```

Comment:
Is the assignment that is happening here providing value? Wondering if we could replace it with `iterator_to_array`.